Index: whowatch/screen.c
diff -u whowatch/screen.c:1.1 whowatch/screen.c:1.2
--- whowatch/screen.c:1.1	Mon Aug 21 08:34:27 2000
+++ whowatch/screen.c	Mon Aug 21 08:38:47 2000
@@ -6,6 +6,7 @@
 struct window info_win;
 static chtype *curs_buf;
 extern int screen_cols;
+int old_cur_vis = 1; /* this is the cursor mode, set to normal as default */
 
 char *help_line[] = 
 	{
@@ -38,7 +39,7 @@
 	}
 
 	wattrset(users_list.wd, A_BOLD);
-        printf("\033[?25l");                    /* disable cursor */
+        old_cur_vis = curs_set(0);                    /* disable cursor */
 
         start_color();
 	init_pair(1,COLOR_CYAN,COLOR_BLACK);
@@ -58,6 +59,7 @@
 	
 	cbreak();
         nodelay(stdscr,TRUE);
+	keypad(info_win.wd, TRUE);
         scrollok(users_list.wd,TRUE);
         noecho();
 }				
@@ -67,7 +69,7 @@
 	werase(help_win.wd);
 	wrefresh(help_win.wd);
 	endwin();
-        printf("\033[?25h");            /* enable cursor */
+        curs_set(old_cur_vis);            /* enable cursor */
 }
 
 void cursor_on(struct window *w, int line)
@@ -116,7 +118,8 @@
 		if (i > w->cols) break;
 		if (*p < 17){
 			i--;
-			waddnstr(w->wd, q, p - q);
+			if (p - q != 0)
+				waddnstr(w->wd, q, p - q);
 			wattrset(w->wd, COLOR_PAIR(*p));
 			q = p + 1;
 		}
Index: whowatch/whowatch.c
diff -u whowatch/whowatch.c:1.1 whowatch/whowatch.c:1.2
--- whowatch/whowatch.c:1.1	Mon Aug 21 08:34:27 2000
+++ whowatch/whowatch.c	Mon Aug 21 08:38:48 2000
@@ -20,8 +20,7 @@
 #define ut_user ut_name
 #endif
 
-enum key {  ENTER=0x100, UP, DOWN, LEFT, RIGHT, DELETE, ESC, CTRL_K, CTRL_I,
-	   PG_DOWN, PG_UP, HOME, END };
+enum key {  ENTER=KEY_MAX + 1, ESC, CTRL_K, CTRL_I};
 	   
 enum State{ USERS_LIST, PROC_TREE, INIT_TREE } state;
 
@@ -334,30 +333,12 @@
 int read_key()
 {
 	int c;
-	c = getc(stdin);
+	c = wgetch(info_win.wd);
 	switch (c){
 		case 0xD:
 		case 0xA: return ENTER;
 		case 0xB: return CTRL_K;
 		case 0x9: return CTRL_I;
-		case 0x1B:
-			getc(stdin);
-			c = getc(stdin);
-			switch(c) {
-				case 0x41: return UP;
-				case 0x42: return DOWN;
-				case 0x34:
-				case 0x38:
-				case 0x46: return END;
-				case 0x36:
-				case 0x47: return PG_DOWN;
-				case 0x31:
-				case 0x37:
-				case 0x48: return HOME;
-				case 0x35:
-				case 0x49: return PG_UP;
-			}
-			break;
 		default:
 			break;
 	}
@@ -420,6 +401,7 @@
 	    signal_sent = 0;
 	}
 	switch(key){
+	case KEY_ENTER:
 	case ENTER:
 		werase(windows[state]->wd);
 		switch(state){
@@ -462,23 +444,23 @@
 		send_signal(9, pid);
 		tree_periodic();
 		break;
-	case PG_DOWN:
+	case KEY_NPAGE:
 		page_down(windows[state], rfrsh[state]);
 		break;
-	case PG_UP:
+	case KEY_PPAGE:
 		page_up(windows[state], rfrsh[state]);
 		break;
-	case HOME:
+	case KEY_HOME:
 		key_home(windows[state], rfrsh[state]);
 		break;
-	case END:
+	case KEY_END:
 		key_end(windows[state], rfrsh[state]);
 		break;
-	case UP:
+	case KEY_UP:
 		cursor_up(windows[state]);
 		wrefresh(windows[state]->wd);
 		break;
-	case DOWN:
+	case KEY_DOWN:
 		cursor_down(windows[state]);
 		wrefresh(windows[state]->wd);
 		break;
Index: whowatch/whowatch.h
diff -u whowatch/whowatch.h:1.1 whowatch/whowatch.h:1.2
--- whowatch/whowatch.h:1.1	Mon Aug 21 08:34:27 2000
+++ whowatch/whowatch.h	Mon Aug 21 08:38:48 2000
@@ -9,7 +9,7 @@
 #include <sys/types.h>
 #include <sys/time.h>
 #include <sys/ioctl.h>
-#include <ncurses.h>
+#include <curses.h>
 #include <assert.h>
 
 #define CURSOR_COLOR	A_REVERSE

-- 
